server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -590,7 +590,14 @@ protected void trimUnreferencedReaders(boolean onlyTrimLocal) throws IOException
             generationsToDelete.add(generation);
         }
         if (generationsToDelete.isEmpty() == false) {
-            deleteRemoteGeneration(generationsToDelete);
+            try {
+                deleteRemoteGeneration(generationsToDelete);
+            } catch (Exception e) {
+                logger.error("Exception in delete generations flow", e);
+                // Release permit that is meant for metadata files and return
+                remoteGenerationDeletionPermits.release();
+                return;
+            }
             translogTransferManager.deleteStaleTranslogMetadataFilesAsync(remoteGenerationDeletionPermits::release);
```

Comment:
Do we need similar handling for this line as done for deleteRemoteGeneration. We wont need to release permits during exception, but for the logs we can have same behaviour of logging and not throwing it to the caller?

Reply:
`deleteStaleTranslogMetadataFilesAsync ` doesn't throw the exception.